Installation issue: embree ^intel-oneapi-tbb@2021.11.0

### Steps to reproduce the issue

```console
docker run --rm -it spack/ubuntu-jammy:develop install embree -ispc ^intel-oneapi-tbb@2021.11.0
```
(`-ispc` solely to avoid building `llvm`)

### Error message

<details><summary>Error message</summary>

<pre>
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/gcc-runtime-11.4.0-vfbxhvvvqok2icbiykzq3fupmz4oizzz
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/nghttp2-1.57.0-25g474kpy6wfl4hayd4q7quoagxgzm2m
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/intel-oneapi-tbb-2021.11.0-5fcqebqcspqhud6jmgp62tke7n5exkw6
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/ncurses-6.4-6mh657cxth45nv553pzrz6ayspfags3s
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/gmake-4.4.1-echlqtvk7cd6yqexjoly3zqwtpogvsqo
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/zlib-ng-2.1.5-xa7ucpzwvnunrt6naeitvgc67m2vhlue
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/openssl-3.1.3-5vqn5um2as5dzz3mbutal63q4nz5owpb
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/curl-8.4.0-ctocybdtyudjdq2wz6ziqabab5u6iv2p
[+] /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/cmake-3.27.9-hlso7rvucd4pfygrg2ps2xqjahqufqbt
==> Installing embree-4.3.0-a2xmwarebryohff63ausyrx3viamfaly [10/10]
==> No binary for embree-4.3.0-a2xmwarebryohff63ausyrx3viamfaly found: installing from source
==> Using cached archive: /opt/spack/var/spack/cache/_source-cache/archive/ba/baf0a57a45837fc055ba828a139467bce0bc0c6a9a5f2dccb05163d012c12308.tar.gz
==> No patches needed for embree
==> embree: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/cmake-3.27.9-hlso7rvucd4pfygrg2ps2xqjahqufqbt/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/embree-4.3.0-a2xmwarebryohff63ausyrx3viamfaly' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/embree-4.3.0-a2xmwarebryohff63ausyrx3viamfaly/lib;/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/embree-4.3.0-a2xmwarebryohff63ausyrx3viamfaly/lib64;/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/gcc-runtime-11.4.0-vfbxhvvvqok2icbiykzq3fupmz4oizzz/lib' '-DCMAKE_PREFIX_PATH:STRING=/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/intel-oneapi-tbb-2021.11.0-5fcqebqcspqhud6jmgp62tke7n5exkw6;/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/cmake-3.27.9-hlso7rvucd4pfygrg2ps2xqjahqufqbt;/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/gmake-4.4.1-echlqtvk7cd6yqexjoly3zqwtpogvsqo;/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/gcc-runtime-11.4.0-vfbxhvvvqok2icbiykzq3fupmz4oizzz' '-DBUILD_TESTING=OFF' '-DEMBREE_TUTORIALS=OFF' '-DEMBREE_IGNORE_CMAKE_CXX_FLAGS=ON' '-DEMBREE_ISPC_SUPPORT:BOOL=OFF' '-DEMBREE_TBB_ROOT:STRING=/opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/intel-oneapi-tbb-2021.11.0-5fcqebqcspqhud6jmgp62tke7n5exkw6' '-DEMBREE_ISA_SSE2:BOOL=OFF' '-DEMBREE_ISA_SSE42:BOOL=OFF' '-DEMBREE_ISA_AVX:BOOL=OFF' '-DEMBREE_ISA_AVX2:BOOL=ON' '-DEMBREE_ISA_AVX512:BOOL=ON' '/tmp/root/spack-stage/spack-stage-embree-4.3.0-a2xmwarebryohff63ausyrx3viamfaly/spack-src'

1 error found in build log:
     35    Call Stack (most recent call first):
     36      common/tasking/CMakeLists.txt:33 (find_package)
     37    
     38    
     39    -- Looking for TBB components tbb; ()
     40    -- Found TBB version 2021.11 at /opt/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.4.0/intel-oneapi-tbb-2021.11.0-5fcqebqcspqhud6jmgp62tke7n5exkw6
  >> 41    CMake Error at common/cmake/FindTBB.cmake:72 (message):
     42      Cannot find required components: ;tbb
     43    Call Stack (most recent call first):
     44      common/cmake/FindTBB.cmake:477 (rk_tbb_error)
     45      common/tasking/CMakeLists.txt:33 (find_package)
     46    
     47    

See build log for details:
  /tmp/root/spack-stage/spack-stage-embree-4.3.0-a2xmwarebryohff63ausyrx3viamfaly/spack-build-out.txt

</pre></details>


### Information on your system

* **Spack:** 0.22.0.dev0 (95390370969c5e12777c4e4e542f5d6dd6ede5dc)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-skylake
* **Concretizer:** clingo


### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/13980871/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/13980872/spack-build-out.txt)

Maintainers: @aumuell 

## Additional diagnosis
This issue does not happen with `spack install embree -ispc ^intel-oneapi-tbb@2021.10.0`.

A workaround is to apply the following diff:
```diff
diff --git a/var/spack/repos/builtin/packages/embree/package.py b/var/spack/repos/builtin/packages/embree/package.py
index 72431fb78a..bc977be18b 100644
--- a/var/spack/repos/builtin/packages/embree/package.py
+++ b/var/spack/repos/builtin/packages/embree/package.py
@@ -59,7 +59,7 @@ def cmake_args(self):
             "-DEMBREE_TUTORIALS=OFF",
             "-DEMBREE_IGNORE_CMAKE_CXX_FLAGS=ON",
             self.define_from_variant("EMBREE_ISPC_SUPPORT", "ispc"),
-            self.define("EMBREE_TBB_ROOT", spec["tbb"].prefix),
+            self.define("EMBREE_TBB_ROOT", spec["tbb"].prefix.tbb.latest),
         ]
 
         if spec.satisfies("target=x86_64:") or spec.satisfies("target=x86:"):
```

This may be caused by 3d744e7c95 ("intel-oneapi 2024.0.0: added new version to packages (#41135)", 2023-11-27), @prstrnn @rscohn2.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/embree/package.py
+++ b/var/spack/repos/builtin/packages/embree/package.py
@@ -59,7 +59,6 @@ class Embree(CMakePackage):
             "-DEMBREE_TUTORIALS=OFF",
             "-DEMBREE_IGNORE_CMAKE_CXX_FLAGS=ON",
             self.define_from_variant("EMBREE_ISPC_SUPPORT", "ispc"),
-            self.define("EMBREE_TBB_ROOT", spec["tbb"].prefix),
         ]
 
         if spec.satisfies("target=x86_64:") or spec.satisfies("target=x86:"):

